### PR TITLE
Document structured APT content

### PR DESCRIPTION
**Pull request by maximiliankolb**
#### What changes are you introducing?

docs for Foreman+Katello to provide structured APT content

#### Why are you introducing these changes? (Explanation, links to references, issues, etc.)

docs for https://github.com/Katello/katello/pull/11058

#### Anything else to add? (Considerations, potential downsides, alternative solutions you have explored, etc.)

the PR is based on a KB article in ATIX Service Portal.

#### Checklists

* [ ] I am okay with my commits getting squashed when you merge this PR.
* [x] I am familiar with the [contributing](https://github.com/theforeman/foreman-documentation/blob/master/CONTRIBUTING.md) guidelines.

Please cherry-pick my commits into:

* [x] Foreman 3.13/Katello 4.15 (Satellite 6.17)
* [ ] Foreman 3.12/Katello 4.14 (Satellite 6.16)
* [ ] Foreman 3.11/Katello 4.13 (orcharhino 6.11 on EL8 only)
* [ ] Foreman 3.10/Katello 4.12
* [ ] Foreman 3.9/Katello 4.11 (Satellite 6.15; orcharhino 6.8/6.9/6.10)
* [ ] Foreman 3.8/Katello 4.10
* [ ] Foreman 3.7/Katello 4.9 (Satellite 6.14)
* [ ] Foreman 3.6/Katello 4.8
* [ ] Foreman 3.5/Katello 4.7 (Satellite 6.13; orcharhino 6.6/6.7)
* We do not accept PRs for Foreman older than 3.5.

### CODE REVIEW

File: guides/common/modules/proc_enabling-structured-apt-content.adoc [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+endif::[]
+ifndef::orcharhino[]
+* Your {DL} host runs `subscription-manager` version 1.29.35-3 or higher.
+endif::[]
+
+.Procedure
+. Backup your {ProjectServer}.
++
+For more information, see {AdministeringDocURL}backing-up-{project-context}-server-and-{smart-proxy-context}_admin[Backing Up {ProjectServer} and {SmartProxyServer}] in _{AdministeringDocTitle}_.
+. Verify that there are no running tasks:
```
Comment (maximiliankolb) [inline guides/common/modules/proc_enabling-structured-apt-content.adoc:21]: alternative: "Ensure ..."?

File: guides/common/modules/proc_enabling-structured-apt-content.adoc [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+# {foreman-maintain} maintenance-mode start
+----
+. Migrate your Deb content:
++
+[options="nowrap" subs="+quotes,verbatim,attributes"]
+----
+# foreman-rake katello:migrate_structure_content_for_deb
+----
++
+On {Project} with 40 Deb repositories published to many content view versions and promoted to multiple lifecycle environments each, this step took 15 minutes.
```
Comment (sbernhard) [inline guides/common/modules/proc_enabling-structured-apt-content.adoc:42]: This amount of  time may differ and depends on packages, repositories, the used environment and other tasks running on your machine.

File: guides/common/modules/proc_enabling-structured-apt-content.adoc (hunk 1 above)
Comment (sbernhard) [reply guides/common/modules/proc_enabling-structured-apt-content.adoc:21]: Ensure is better

Comment (maximiliankolb) [reply guides/common/modules/proc_enabling-structured-apt-content.adoc]: Applied.

File: guides/common/modules/proc_enabling-structured-apt-content.adoc (hunk 2 above)
Comment (maximiliankolb) [reply guides/common/modules/proc_enabling-structured-apt-content.adoc:42]: Slightly reworded and applied.

**Review by maximiliankolb — COMMENTED** [review]
Applied both suggestions.

File: guides/common/modules/con_using-structured-apt-content.adoc [hunk 3]
```diff
@@ -0,0 +1,32 @@
+[id="using-structured-apt-content"]
+= Using structured APT content
```
Comment (Lennonka) [inline guides/common/modules/con_using-structured-apt-content.adoc:2]: This isn't a procedure, so I'd avoid the gerundium.
```suggestion
= Structured APT content
```

File: guides/common/modules/con_using-structured-apt-content.adoc [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+
+You can enable structured APT content on your {Project} to use the same Deb repository structure as the remote repositories they are synchronized from.
+
+Features::
```
Comment (Lennonka) [inline guides/common/modules/con_using-structured-apt-content.adoc:6]: I'd drop this to simplify it. The text looks rather repetitive.

File: guides/common/modules/con_using-structured-apt-content.adoc [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+With structured APT content, Deb repositories on {Project} use the same structure as the upstream Deb repository.
+This allows features that attach to the repository structure to work the same way as they would with the upstream repository, for example:
+
+* `apt list --upgradable` will no longer list `unknown,unknown` as the release and distribution for packages.
+Instead, `apt` will list the proper release and distribution from the upstream repository.
+* You can use client-side APT pinning in the same way as when using the upstream repository directly.
+
+Comprehensibility::
+With structured APT, repositories in {Project} simply use the same structure as the upstream repository.
+They do not use the special `default/all` simple publication concept which does not exist outside of {Project}.
```
Comment (Lennonka) [inline guides/common/modules/con_using-structured-apt-content.adoc:16]: The structuring is good but I'd use the same formatting for these bits as for the Limitations below.

```suggestion
.Comprehensibility
With structured APT, repositories in {Project} use the same structure as the upstream repository.
They do not use the special `default/all` publication concept which does not exist outside of {Project}.
```
Drop "simply". I'm not sure what the last sentence means but it looks overburdened with adjectives.

File: guides/common/modules/con_using-structured-apt-content.adoc [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+
+Performance::
+With structured APT content enabled, publishing repository metadata will take roughly half the time, compared to with structured APT disabled.
```
Comment (Lennonka) [inline guides/common/modules/con_using-structured-apt-content.adoc:19]:
```suggestion
.Performance
With structured APT content, publishing repository metadata will take roughly half the time, compared to publishing without structured APT content.
```
Trying to improve readability.

File: guides/common/modules/con_using-structured-apt-content.adoc [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+
+.Limitations
+* Once you have enabled structured APT on {Project}, you should not disable it.
```
Comment (Lennonka) [inline guides/common/modules/con_using-structured-apt-content.adoc:22]:
```suggestion
* Once you have enabled structured APT on {Project}, do not disable it.
```
We shouldn't use "should" :) Do or do not, can, cannot, have to, must not...

File: guides/common/modules/proc_enabling-structured-apt-content.adoc [hunk 8]
```diff
@@ -0,0 +1,52 @@
+[id="enabling-structured-apt-content"]
+= Enabling structured APT content
+
+You can enable structured APT content on your {Project} to use the same Deb repository structure as the remote repositories they are synchronized from.
+
+.Prerequisites
+* The host is registered to {ProjectName}.
```
Comment (Lennonka) [inline guides/common/modules/proc_enabling-structured-apt-content.adoc:7]: What host? I thought we're performing this procedure on ProjectServer. This confuses me.

File: guides/common/modules/proc_enabling-structured-apt-content.adoc [hunk 9]
```diff
@@ -0,0 +1,52 @@
...
+include::snip_prerequisite-project-client-repository-enabled.adoc[]
```
Comment (Lennonka) [inline guides/common/modules/proc_enabling-structured-apt-content.adoc:8]: What do you need the Client repo for? Again, confused. It isn't clear from the procedure.

**Review by Lennonka — CHANGES_REQUESTED** [review]
A couple of questions:
- Wouldn't this work better in the Administering guide? The setting should be added to the Content settings reference, at least.
- Instead of "enabling structured APT content", wouldn't you rather say "migrating X to structured APT content"? Migration is usually one-time and one-way thing, but enabling suggests that it can be disabled again.

File: guides/common/modules/con_using-structured-apt-content.adoc [hunk 10]
```diff
@@ -0,0 +1,32 @@
...
+Features::
+With structured APT content, Deb repositories on {Project} use the same structure as the upstream Deb repository.
+This allows features that attach to the repository structure to work the same way as they would with the upstream repository, for example:
+
+* `apt list --upgradable` will no longer list `unknown,unknown` as the release and distribution for packages.
+Instead, `apt` will list the proper release and distribution from the upstream repository.
+* You can use client-side APT pinning in the same way as when using the upstream repository directly.
+
+Comprehensibility::
+With structured APT, repositories in {Project} simply use the same structure as the upstream repository.
```
Comment (Lennonka) [inline guides/common/modules/con_using-structured-apt-content.adoc:15]:
```suggestion
With structured APT content, repositories in {Project} simply use the same structure as the upstream repository.
```

File: guides/common/modules/proc_enabling-structured-apt-content.adoc (hunk 8 above)
Comment (maximiliankolb) [reply guides/common/modules/proc_enabling-structured-apt-content.adoc]: It's a prerequisite for hosts that consume structured APT content from Foreman+Katello. -> Moved.

**Review by maximiliankolb — COMMENTED** [review]
Rebased to "master" and applied all feedback. I hope it's ok to list the prerequisites for hosts that want to consume structured APT content in the concept file. What do you think @Lennonka ?

**Review by Lennonka — APPROVED** [review]
Yes, that works nicely! LGTM

**Comment by maximiliankolb** [comment]
Merged to "master" and cherry-picked to "3.13":
   472787ccd0..ce4dea0489  3.13 -> 3.13

**Pull request merged**